tests/test_reward_trainer.py

```diff
@@ -14,159 +14,111 @@
 import tempfile
 import unittest
 
-import pytest
 import torch
-from datasets import Dataset
+from datasets import Dataset, load_dataset
 from transformers import AutoModelForSequenceClassification, AutoTokenizer, EvalPrediction
 from transformers.testing_utils import require_peft
+from transformers.utils import is_peft_available
 
-from trl import RewardConfig, RewardTrainer
+from trl import RewardConfig, RewardTrainer, maybe_apply_chat_template
 from trl.trainer import compute_accuracy
+from trl.trainer.reward_trainer import _tokenize
+
+
+if is_peft_available():
+    from peft import LoraConfig, TaskType
 
 
 class RewardTrainerTester(unittest.TestCase):
+    def setUp(self):
+        self.model_id = "hf-internal-testing/tiny-random-LlamaForCausalLM"
```

Comment:
I think we should gradually move towards testing the most popular LLM architectures instead of relying on `gpt2` which has a bunch of annoying things like a missing PAD token